packages/lesswrong/components/editor/RateLimitWarning.tsx

```diff
@@ -3,21 +3,23 @@ import { Components, registerComponent } from '../../lib/vulcan-lib';
 import moment from 'moment';
 import { isEAForum } from '../../lib/instanceSettings';
 
-const RateLimitWarning = ({lastRateLimitExpiry}: {
+// Tells the user when they can next comment or post if they're rate limited, and a brief explanation
+const RateLimitWarning = ({lastRateLimitExpiry, rateLimitMessage}: {
   lastRateLimitExpiry?: Date|null,
+  rateLimitMessage?: string|null
 }) => {
   // Sorry this is not great. Basically, I wanted to keep the previous functionality for the default case
   // (which shows how long you have until you can comment again), and then on the EA Forum, show the
   // particular copy that our product team wants. In the future we probably want to pass in the reason
   // that the user is currently rate-limited, to display the appropriate message.
-  if (!isEAForum && !lastRateLimitExpiry) return null
```

Comment:
I'm quite confused about what's going on here – seems like either lastRateLimitExpiry should be required (and this null-return handled in the parent component) or I'd expect EAForum to also want to return null here. What's going on with this chesterton fence?

Reply:
Sorry, the copy changed after I wrote that comment. You can see the latest [here](https://github.com/ForumMagnum/ForumMagnum/pull/7128/files#diff-e7d7df813ced31a065a267f489d6e4b228ebc2054aa3059c9fc6a08e10141fd5)